rpc/src/v1/impls/chain.rs

```diff
@@ -124,16 +124,20 @@ where
         shard_id: ShardId,
         block_number: Option<u64>,
     ) -> Result<Option<AssetScheme>> {
-        let address = AssetSchemeAddress::new(tracker, shard_id);
-        self.get_asset_scheme_by_type(address.into(), block_number)
+        let block_id = block_number.map(BlockId::Number).unwrap_or(BlockId::Latest);
+        let address = AssetSchemeAddress::new_from_tracker(tracker, shard_id);
+        self.client.get_asset_scheme(address, block_id).map_err(errors::transaction_state)
```

Comment:
How about making `get_asset_scheme` receive the `asset_type` and `shard_id` instead of the asset scheme address?

Reply:
I think it's a good idea. Also I'll look at other files using `AssetSchemeAddress`.

Reply:
Now `AssetSchemeAddress` is only used in codechain-state. I'll delete it from the SDK too.